pysat/_constellation.py

```diff
@@ -108,10 +140,121 @@ def __str__(self):
                 output_str += "{:d} '{:s}' '{:s}' '{:s}' '{:s}'\n".format(
                     i, inst.platform, inst.name, inst.tag, inst.inst_id)
         else:
-            output_str += "No loaded Instruments\n"
+            output_str += "No assigned Instruments\n"
+
+        # Display loaded data
+        output_str += '\n\nLoaded Data Statistics\n'
+        output_str += '----------------------\n'
+        if not self.empty:
+            output_str += 'Date: ' + self.date.strftime('%d %B %Y') + '\n'
+            output_str += '{:s}time range: '.format(
+                "Common " if self.common_index else "Full ")
+            output_str += self.index[0].strftime('%d %B %Y %H:%M:%S')
+            output_str += ' --- '
+            output_str += self.index[-1].strftime('%d %B %Y %H:%M:%S\n')
+            output_str += 'Number of variables: {:d}\n'.format(
+                len(self.variables))
+
+            output_str += '\nVariable Names:\n'
+            output_str += utils._core.fmt_output_in_cols(self.variables)
+        else:
+            output_str += 'No loaded data.\n'
 
         return output_str
 
+    # -----------------------------------------------------------------------
+    # Define all hidden methods
+
+    def _empty(self, all_inst=True):
+        """Boolean flag reflecting lack of data
+
+        Parameters
+        ----------
+        all_inst : bool
+            Require all instruments to have data for the emtpy flag to be
+            False, if True.  If False, the empty flag will be False if any
+            instrument has data. (default=True)
+
+        Returns
+        -------
+        eflag : bool
+            True if there is no Instrument data, False if there is data
+            in at least one/all Instruments.
+
+        """
+        eflags = [inst.empty for inst in self.instruments]
+
+        if len(eflags) == 0:
+            eflag = True
+        elif all_inst:
+            eflag = np.any(eflags)
+        else:
+            eflag = np.all(eflags)
+
+        return eflag
+
+    def _index(self):
+        """Returns a common time index for the loaded data
```

Comment:
For something like ICON which is never `full` I'd still like to be able to use the common index. I get that if the index covers times where everything has data, a single instrument without means no common area. But again, ICON.

Brainstorming idea:
What if there was a strict option? If strict, the do current behavior. If not strict, index covers common range of times for instruments with data.

I think there could be/are problems with my suggestion. I'll keep thinking myself.